Tests/CorePromise/WhenResolvedTests.swift

```diff
@@ -5,11 +5,12 @@ import PromiseKit
 import XCTest
 
 class JoinTests: XCTestCase {
+    
     func testImmediates() {
         let successPromise = Promise()
 
         var joinFinished = false
-        when(resolved: successPromise).done(on: nil) { _ in joinFinished = true }
+        when(resolved: [successPromise]).done(on: nil) { joinFinished = true }
```

Comment:
So, does this mean we have broken API? We can't break API.

Reply:
This is because of "ambiguous use of 'when'" errors. I have commented out the responsible functions for now:
https://github.com/Skoti/PromiseKit/blob/d13d220396cf88d649d02ef82549eeb85cc95816/Sources/when.swift#L86-L89
and here:
https://github.com/Skoti/PromiseKit/blob/d13d220396cf88d649d02ef82549eeb85cc95816/Sources/when.swift#L302-L305
Left them commented out for a reminder.

It seems like adding this `R` associated type representing result in `Thenable` protocol confuses the compiler for some reason? 🤷‍♂ As before that it was not a problem.

Anyway, I think tuple overloads with this `R` associated type give us much more flexibility of using Promises and Guarantees as well as mixing them and a single use case of a convenience `Void` variadic function is not taking these advantages down. Plus it is a really easy fix for such a breaking change - you just add `[` and `]` around the arguments...

Reply:
I agree this is better, but breaking API is not allowed, API breakage can only happen for major version bumps, so this may have to be a v7 PR if we cannot solve it.

Reply:
Ok! Just let me know when you are ready with the v7 changes. Then I will rebase this PR and you can review it and possibly merge before v7 release 😄

Reply:
v7 is ready for submissions.

Reply:
hmm but the v7 branch was updated 6-7 months ago, is this is the one I should use?